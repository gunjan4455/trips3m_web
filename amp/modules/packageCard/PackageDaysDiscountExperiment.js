import React from 'react';
import PropTypes from 'prop-types';

import './PackageDaysDiscount.scss';

const PackageDaysDiscountExperiment = ({ days, nights, isPackageRevamp}) => (
  <div className="row row-">
    <p className={`at_packageDaysCount fw7 text-left ${isPackageRevamp ? 'f10 at_package' : 'f12'}`}>{days} Days & {nights} Nights </p>
  </div>
);

PackageDaysDiscountExperiment.propTypes = {
  days: PropTypes.number.isRequired,
  nights: PropTypes.number.isRequired,
  isPackageRevamp: PropTypes.bool
};

export default PackageDaysDiscountExperiment;
