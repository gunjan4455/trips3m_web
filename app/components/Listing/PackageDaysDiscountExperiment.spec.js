import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import PackageDaysDiscountExperiment from './PackageDaysDiscountExperiment';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  days: 6,
  nights: 5,
};

const shallowWrapperPackageDaysDiscountExperiment = (props) => {
  return shallow(<PackageDaysDiscountExperiment {...props} />);
};


describe('Package card module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperPackageDaysDiscountExperiment(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render at_package when isPackageRevamp is undefined', () => {
    const wrapper = shallowWrapperPackageDaysDiscountExperiment(props);
    expect(wrapper.find('.at_package')).toHaveLength(0);
  });

  it('should render at_package when isPackageRevamp is true', () => {
    props.isPackageRevamp = true;
    const wrapper = shallowWrapperPackageDaysDiscountExperiment(props);
    expect(wrapper.find('.at_package')).toHaveLength(1);
  })

});
