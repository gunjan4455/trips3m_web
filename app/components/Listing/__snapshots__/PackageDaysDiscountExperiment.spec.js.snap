// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package card module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PackageDaysDiscountExperiment
    days={6}
    nights={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="at_packageDaysCount fw7 text-left f12"
      >
        6
         Days & 
        5
         Nights 
      </p>,
      "className": "row row-",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          6,
          " Days & ",
          5,
          " Nights ",
        ],
        "className": "at_packageDaysCount fw7 text-left f12",
      },
      "ref": null,
      "rendered": Array [
        6,
        " Days & ",
        5,
        " Nights ",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="at_packageDaysCount fw7 text-left f12"
        >
          6
           Days & 
          5
           Nights 
        </p>,
        "className": "row row-",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            6,
            " Days & ",
            5,
            " Nights ",
          ],
          "className": "at_packageDaysCount fw7 text-left f12",
        },
        "ref": null,
        "rendered": Array [
          6,
          " Days & ",
          5,
          " Nights ",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
